### PR TITLE
Jsfix

### LINKED ISSUE #243: Folder Exclusion / Inclusion options not displaying properly
image

### PR DESCRIPTION
**Pull request by jamesros161**
This resolves #243

### CODE REVIEW

File: admin/js/boldgrid-backup-admin-folder-exclude.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ BoldGrid.FolderExclude = function( $ ) {
 		exclusionList = null,
 		filteredList = [],
 		lang = BoldGridBackupAdminFolderExclude,
-		$container = $( '#folder_exclusion' ),
+		$container = $( 'div#folder_exclusion' ),
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-folder-exclude.js:22]: We we need to have the selector this specific?

File: admin/js/boldgrid-backup-admin-folder-exclude.js [hunk 2]
```diff
@@ -161,8 +161,8 @@ BoldGrid.FolderExclude = function( $ ) {
 	 *
 	 * @since 1.6.0
 	 */
-	self.onChangeType = function() {
-		self.toggleConfig();
+	self.onChangeType = function( type ) {
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-folder-exclude.js:164]: Need to add `type` to the docblock.

File: admin/js/boldgrid-backup-admin-folder-exclude.js [hunk 3]
```diff
@@ -352,14 +352,16 @@ BoldGrid.FolderExclude = function( $ ) {
 	 *
 	 * @since 1.6.0
 	 */
-	self.toggleConfig = function() {
-		var type = $type.filter( ':checked' ).val(),
-			$miscInfo = $( '#folder_misc_info' );
+	self.toggleConfig = function( typeInput ) {
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-folder-exclude.js:355]: `typeInput` Needs to be added to the docblock.

File: admin/js/boldgrid-backup-admin-folder-exclude.js [hunk 4]
```diff
@@ -352,14 +352,16 @@ BoldGrid.FolderExclude = function( $ ) {
...
+		var type = $( typeInput )
+			.filter( ':checked' )
+			.val();
+		var $miscInfo = $( 'div#folder_misc_info' );
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-folder-exclude.js:359]: We can put everything in one `var` statement - https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#declaring-variables-with-var

File: admin/js/boldgrid-backup-admin-folder-exclude.js [hunk 5]
```diff
@@ -400,9 +402,13 @@ BoldGrid.FolderExclude = function( $ ) {
...
 
 		self.toggleStatus();
-		self.toggleConfig();
+		$type.each( function() {
+			self.toggleConfig( this );
+		} );
 
-		$type.on( 'change', self.onChangeType );
+		$type.on( 'change', function() {
+			self.onChangeType( this );
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-folder-exclude.js:410]: Do you actually need to make this change? `this` is already passed / available in the `onChangeType` method.

File: admin/js/boldgrid-backup-admin-settings-autoupdate.js [hunk 6]
```diff
@@ -53,7 +53,9 @@ BOLDGRID.SETTINGS = BOLDGRID.SETTINGS || {};
 			$bgBox
 				.find( '.table-help td p' )
 				.attr( 'style', 'height: 0em; opacity: 0%; position: relative; z-index:-1' );
-			$bgBox.find( '.dashicons-editor-help, .help-icon' ).on( 'click', self._toggleHelp );
+			$bgBox
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-settings-autoupdate.js:56]: Is this change related to the issue? Did you intend to add this?

File: admin/js/boldgrid-backup-admin-table-includes.js [hunk 7]
```diff
@@ -16,7 +16,7 @@ BoldGrid.TableInclude = function( $ ) {
 	'use strict';
 
 	var self = this,
-		$container = $( '#table_inclusion' ),
+		$container = $( 'div#table_inclusion' ),
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-table-includes.js:19]: All the changes you made to boldgrid-backup-admin-folder-exclude.js where made to this file as well. Please see those comments.

File: admin/js/boldgrid-backup-admin-folder-exclude.js (hunk 1 above)
Comment (jamesros161) [reply admin/js/boldgrid-backup-admin-folder-exclude.js:22]: Yes. In fact, the reason why we need to have teh div#folder_exclusion is because of the duplicate id's on the page. This is a result of the folder_exclusion element id's being present in the settings page, as well as in the Backup Now Modal. In order for them to both work, then the div#folder_exclusion must be present. This causes jQuery to treat the result as a list of elements, instead of picking the first one it finds.

Comment (jamesros161) [reply admin/js/boldgrid-backup-admin-folder-exclude.js:22]: This is actually the solution suggested by Rafael

File: admin/js/boldgrid-backup-admin-folder-exclude.js (hunk 5 above)
Comment (jamesros161) [reply admin/js/boldgrid-backup-admin-folder-exclude.js:412]: Not quite.....If you don't change it, the variable passed is the event object, wheras in this situation, the variable passed is the element that changed.

image

File: admin/js/boldgrid-backup-admin-settings-autoupdate.js (hunk 6 above)
Comment (jamesros161) [reply admin/js/boldgrid-backup-admin-settings-autoupdate.js:56]: This is to resolve a different but related bug that there was no issue for. The help icons were not working in some places as a result of my code for the ones within the table body being done differently. I had to increase the specificity of the selector for it, so that my code wasn't effecting all the other help icons on the settings page.

File: admin/js/boldgrid-backup-admin-table-includes.js (hunk 7 above)
Comment (jamesros161) [reply admin/js/boldgrid-backup-admin-table-includes.js:19]: See the above comments

**Comment by jamesros161** [comment]
Requested Changes Made

**Comment by bwmarkle** [comment]
image

File: admin/js/boldgrid-backup-admin-folder-exclude.js [hunk 8]
```diff
@@ -374,7 +374,16 @@ BoldGrid.FolderExclude = function( $ ) {
...
-	self.toggleStatus = function() {
+	self.toggleStatus = function( eventTarget ) {
+		var parentContainer;
+		if ( eventTarget ) {
+			parentContainer = eventTarget.closest( 'div#folder_exclusion' );
+			$inputInclude = $( parentContainer ).find( 'input[name=folder_exclusion_include]' );
+			$inputExclude = $( parentContainer ).find( 'input[name=folder_exclusion_exclude]' );
+		}
+
+		console.log( $inputInclude );
```
Comment (bwmarkle) [inline admin/js/boldgrid-backup-admin-folder-exclude.js:389]: Can remove console logging

**Comment by jamesros161** [comment]
> 
> 
> image

I have fixed this now.
